### PR TITLE
S5 hints

### PR DESCRIPTION
**Pull request by smccsw**
Story 5 review please

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -79,7 +98,14 @@ document.getElementById('start_button').addEventListener('click', (e) => {
...
+    document.getElementById('reveal_button').disabled = true
     let submit_button = document.getElementById('submit_button')
-    submit_button.disabled = true;
-    submit_button.style.cursor = 'default'
+    submit_button.disabled = true
+})
+
+document.getElementById('hint_button').addEventListener('click', (e) => {
+    document.getElementById('hint_button').disabled = true
+    document.getElementById('hint').style.display = 'block'
```
Comment (ashleycoles) [inline index.js]: This is being repeated now, once in the event listener and then again in get_hint(). Suggest removing it from here and keeping the one in the get_hint() function

File: index.js [hunk 2]
```diff
@@ -1,5 +1,3 @@
-document.getElementById('end_screen').style.display = 'none'
-document.getElementById('game_screen').style.display = 'none'
```
Comment (ashleycoles) [inline index.js]: Remove the unneeded whitespace here

File: index.js [hunk 3]
```diff
@@ -20,25 +18,46 @@ function get_new_question() {
         let quote_element = document.getElementById('quote')
         quote_element.textContent = current_movie.quote
         let answer_element = document.getElementById('answer')
+        answer_element.dataset.year = current_movie.year
+        answer_element.dataset.title = current_movie.title
         answer_element.style.display = 'none'
         answer_element.innerHTML = '<h2>' + current_movie.title + '</h2>'
```
Comment (ashleycoles) [inline index.js:23]: Is this still needed now that you're storing the answer in the dataset? Seems like you're now keeping track of the answer in two places.

File: index.js [hunk 4]
```diff
@@ -20,25 +18,46 @@ function get_new_question() {
...
+        .then((data) => {
+            let answer_year = document.getElementById('answer').dataset.year
+            let answer_title = document.getElementById('answer').dataset.title
+            let movie_titles = (data.films).map(movie => movie.title)
+            let filtered_movies =  movie_titles.filter(movie => movie !== answer_title)
+            let hint = shuffle_array(filtered_movies).slice(0, 2)
+            hint.push(answer_title)
+            shuffle_array(hint)
+            hint.push(answer_year)
+            document.getElementById('hint').innerHTML = '<p>This Movie was released in: ' + hint[3] + '</p> <ul><li>' + hint[0] + '</li><li>' + hint[1] + '</li><li>' + hint[2] + '</li></ul>'
```
Comment (ashleycoles) [inline index.js]: I don't love how this array is storing different things, it relies on you knowing that the key of 3 contains the year. Would be better to use an object with keys instead to make this easier to understand/work on in future

**Review by ashleycoles — CHANGES_REQUESTED** [review]

File: index.js (hunk 3 above)
Comment (Freschwamm) [reply index.js:23]: Yes we believe we do

**Review by ashleycoles — APPROVED** [review]

**Pull request merged**